performance-metrics/src/performance_tests.rs

```diff
@@ -406,6 +424,125 @@ pub fn performance_block_io(control: &PerformanceTestControl) -> f64 {
     }
 }
 
+// Parse the event_monitor file based on the format that each event
+// is followed by a double newline
+fn parse_event_file(event_file: &str) -> Vec<serde_json::Value> {
+    let content = fs::read(event_file).unwrap();
+    let mut ret = Vec::new();
+    for entry in String::from_utf8_lossy(&content)
+        .trim()
+        .split("\n\n")
+        .collect::<Vec<&str>>()
+    {
+        ret.push(serde_json::from_str(entry).unwrap());
+    }
+    ret
+}
+
+fn parse_restore_time_output(events: &[serde_json::Value]) -> Result<f64, Error> {
+    for entry in events.iter() {
+        if entry["event"].as_str().unwrap() == "restored" {
+            let duration = entry["timestamp"]["secs"].as_u64().unwrap() as f64 * 1_000f64
+                + entry["timestamp"]["nanos"].as_u64().unwrap() as f64 / 1_000_000f64;
+            return Ok(duration);
+        }
+    }
+    Err(Error::BootTimeParse)
+}
+
+fn measure_restore_time(
+    cmd: &mut GuestCommand,
+    event_file: &str,
+    test_timeout: u32,
+) -> Result<f64, Error> {
+    let mut child = cmd
+        .capture_output()
+        .verbosity(VerbosityLevel::Warn)
+        .set_print_cmd(false)
+        .spawn()
+        .unwrap();
+
+    thread::sleep(Duration::new(test_timeout as u64, 0));
+    let _ = child.kill();
+    let output = child.wait_with_output().unwrap();
+
+    let json_events = parse_event_file(event_file);
+
+    parse_restore_time_output(&json_events).map_err(|e| {
+        eprintln!(
+            "\n\n==== Start child stdout ====\n\n{}\n\n==== End child stdout ====",
+            String::from_utf8_lossy(&output.stdout)
+        );
+        eprintln!(
+            "\n\n==== Start child stderr ====\n\n{}\n\n==== End child stderr ====",
+            String::from_utf8_lossy(&output.stderr)
+        );
```

Comment:
I think it would be nice to refactor this out (I know there are other parts of this file that do the same.)

Reply:
I try to merge them into one line.